src/libopensc/card-iasecc.c

```diff
@@ -620,6 +668,10 @@ iasecc_init(struct sc_card *card)
 		rv = iasecc_init_amos_or_sagem(card);
 	else if (card->type == SC_CARD_TYPE_IASECC_MI)
 		rv = iasecc_init_amos_or_sagem(card);
+	else if (card->type == SC_CARD_TYPE_IASECC_CPX)
+		rv = iasecc_init_cpx(card);
+	else if (card->type == SC_CARD_TYPE_IASECC_CPXCL)
+		rv = iasecc_init_cpx(card);
```

Comment:
I see this fillows the previous  branches, but I would rather see it in one block such as follows so one does not have to investigate what is different in each of the blocks.

```suggestion
	else if (card->type == SC_CARD_TYPE_IASECC_CPX ||
		  card->type == SC_CARD_TYPE_IASECC_CPXCL)
		rv = iasecc_init_cpx(card);
```
Maybe even better would be the use of `switch`/`case`.

Reply:
I agree that a `switch`/`case` would be a better design, but I do not want to "change" the design logic. It was already a `if`/`else` chain. Once this pull request will be merged, I can provide an update that replaces this `if`/`else` chain with a proper `switch`/`case`.